x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -542,7 +542,7 @@ class AgentPolicyService {
         if (options.withAgentCount) {
           await getAgentsByKuery(
             appContextService.getInternalUserESClient(),
-            appContextService.getInternalUserSOClientForSpaceId(agentPolicy.space_id),
+            appContextService.getInternalUserSOClientForSpaceId(agentPolicy.space_ids?.[0]),
```

Comment:
I'm wondering what is the reason of having a `space_ids` list instead of `space_id`, do we plan to support multiple spaces?
If it can be multiple, should we query agents from all spaces here?

Reply:
@juliaElastic Yes we are going to support multiple space the related PR https://github.com/elastic/kibana/pull/189387  

> should we query agents from all spaces here? 

Yes we should I will update the code